test/box/net.box_reuse_tuple_formats_gh-6217.test.lua

```diff
@@ -0,0 +1,26 @@
+test_run = require('test_run').new()
+net = require('net.box')
+errinj = box.error.injection
+
+box.schema.user.grant('guest', 'execute', 'universe')
+
+-- Check that formats created by net.box for schema are reused (gh-6217).
+MAX_FORMAT_COUNT = 100
+test_run:cmd("setopt delimiter ';'")
+function test()
+    for i = 1, MAX_FORMAT_COUNT do
+        local conn = net.connect(box.cfg.listen)
```

Comment:
Thanks for your patch.

I guess this issue is relevant for huge vshard-clusters. When router has a lot of connections to different storages with the same schema.

E.g. we have a cluster with 1000 instances and 700 spaces on each instance. 1000 * 70 > 65k and this caused a problems. Should it be fixed in scope of this patch? And should we cover this case in tests?

Reply:
It should. I patched the test so that it should cover this case (now I don't create connections immediately).